### PR TITLE
Added Italian and Portuguese translations + Updated version.

### PR DESCRIPTION
**Pull request by kakashi215**
## Hacktoberfest notes:

- due to volume of submissions, we may not be able to review PRs that do not pass tests and do not have informative titles.
- please read our [contributing guidelines](/CONTRIBUTING.md)
- be sure to check the output of Travis-CI for linter errors
- if this is your first open source contribution, make sure it's not your last!

## What does this PR do?
Add Resources | Improve Repo

### Description 
Updated the link of Think Python to the latest edition.
Added two translations of the Python for Everybody book.

### Why is this valuable (or not)?
Extremely valuable update since Think Python 2e uses python3.

### How do we know it's really free?
Written on the respective websites.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book.

### Checklist:
- [x] Not a duplicate
- [x] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [x] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -1588,6 +1588,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [The Java Tutorials](https://docs.oracle.com/javase/tutorial/index.html)
 * [The Java Web Scraping Handbook](https://www.scrapingbee.com/java-webscraping-book) - Kevin Sahin (PDF, HTML)
 * [Think Java: How to Think Like a Computer Scientist](http://greenteapress.com/thinkapjava/) - Allen B. Downey and Chris Mayfield
+* [Think Data Structures: Algorithms and Information Retrieval in Java](https://greenteapress.com/wp/think-data-structures/) - Allen B. Downey (PDF, HTML)
```
Comment (kadhirash) [inline free-programming-books.md:1592]:
```suggestion
* [Think Data Structures: Algorithms and Information Retrieval in Java](https://greenteapress.com/wp/think-data-structures/) - Allen B. Downey (PDF, HTML)
[Think Java: How to Think Like a Computer Scientist](http://greenteapress.com/thinkapjava/) - Allen B. Downey and Chris Mayfield
```

Hello, swap the order to make it alphabetical and the Travis CI build will pass for this commit.

Refer to [CONTRIBUTING](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)

File: free-programming-books.md [hunk 2]
```diff
@@ -2362,6 +2362,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [The Standard Python Library](http://effbot.org/librarybook/) - Fredrik Lundh
 * [Think Complexity](https://greenteapress.com/wp/think-complexity-2e/) - Allen B. Downey (2nd Edition) (PDF, HTML)
 * [Tiny Python 3.6 Notebook](https://github.com/mattharrison/Tiny-Python-3.6-Notebook) - Matt Harrison (3.6)
+* [Think DSP - Digital Signal Processing in Python](https://greenteapress.com/wp/think-dsp/) - Allen B. Downey (PDF, HTML)
```
Comment (kadhirash) [inline free-programming-books.md]: Similar reasoning,

```suggestion
[Think Complexity](https://greenteapress.com/wp/think-complexity-2e/) - Allen B. Downey (2nd Edition) (PDF, HTML)
* [Think DSP - Digital Signal Processing in Python](https://greenteapress.com/wp/think-dsp/) - Allen B. Downey (PDF, HTML)
* [Tiny Python 3.6 Notebook](https://github.com/mattharrison/Tiny-Python-3.6-Notebook) - Matt Harrison (3.6)
```

Swapped the order to make them alphabetical

File: free-programming-books-pt_PT.md [hunk 3]
```diff
@@ -32,3 +33,8 @@
 ### Prolog
 
 * [Lógica Computacional (com Prolog)](http://www4.di.uminho.pt/~mjf/pub/LC-Prolog.pdf) - Maria João Frade (PDF)
+
+
+### Python
+
+* [Python Para Todos: Explorando Dados com Python 3](http://do1.dr-chuck.com/pythonlearn/PT_br/pythonlearn.pdf) - Dr. Charles Russell Severance (PDF) [(EPUB)](http://do1.dr-chuck.com/pythonlearn/PT_br/pythonlearn.epub)
```
Comment (kadhirash) [inline free-programming-books-pt_PT.md:40]: Missing new line at the end of this resource.

> 1 empty line at the end of each .md file

as per the contributing guidelines.

File: free-programming-books.md (hunk 1 above)
Comment (kakashi215) [reply free-programming-books.md]: Thanks!!

File: free-programming-books.md (hunk 2 above)
Comment (kakashi215) [reply free-programming-books.md]: Thanks!!!

File: free-programming-books-pt_PT.md (hunk 3 above)
Comment (kakashi215) [reply free-programming-books-pt_PT.md]: Thanks!!!

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by kakashi215** [comment]
> Thanks, and welcome to free-programming-books!

Thank you so much!!! Looking forward to contribute more.

**Pull request merged**